pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -240,109 +241,220 @@ func substitueYamlVariables(baseYaml []byte, yamlSubstitutions map[string]string
+			manifestLabels := deploymentManifest.GetLabels()
+			if manifestLabels != nil {
+				for key, value := range labels {
+					manifestLabels[key] = value
+				}
+				deploymentManifest.SetLabels(manifestLabels)
+			} else {
+				deploymentManifest.SetLabels(labels)
+			}
+
+			// Check to see whether deployed resource already exists. If not, create else update
+			instanceFound := false
+			list, err := a.Client.DynamicClient.Resource(gvr).Namespace(namespace).List(metav1.ListOptions{})
+			if list != nil && len(list.Items) > 0 {
+				for _, item := range list.Items {
+					if item.GetName() == applicationName {
+						deploymentManifest.SetResourceVersion(item.GetResourceVersion())
+						deploymentManifest.SetAnnotations(item.GetAnnotations())
+						if item.GetKind() == "Service" {
+							currentServiceSpec := item.UnstructuredContent()["spec"].(map[string]interface{})
+							if currentServiceSpec["type"] == "ClusterIP" {
+								newService := deploymentManifest.UnstructuredContent()
+								newService["spec"].(map[string]interface{})["clusterIP"] = currentServiceSpec["clusterIP"]
+								deploymentManifest.SetUnstructuredContent(newService)
+							}
+						}
+						instanceFound = true
+					}
+				}
+			}
+
+			s := log.Spinnerf("Deploying the manifest for %s", gvk.Kind)
+			result := &unstructured.Unstructured{}
+			actionType := "create"
+			if !instanceFound {
+				result, err = a.Client.DynamicClient.Resource(gvr).Namespace(namespace).Create(deploymentManifest, metav1.CreateOptions{})
+			} else {
+				actionType = "update" // Update deployment
+				result, err = a.Client.DynamicClient.Resource(gvr).Namespace(namespace).Update(deploymentManifest, metav1.UpdateOptions{})
+			}
+			if err != nil {
+				s.End(false)
+				return errors.Wrapf(err, "Failed to %s manifest %s", actionType, gvk.Kind)
+			} else {
+				s.End(true)
+				log.Successf("%sd manifest for %s (%s)", strings.Title(actionType), applicationName, gvk.Kind)
 			}
+
+			// Write the returned manifest to the local manifest file
+			if writtenToManifest {
+				_, err = manifestFile.WriteString("---\n")
+				if err != nil {
+					return errors.Wrap(err, "Unable to write to local manifest file")
+				}
+			}
+			err = yamlDecoder.Encode(result, manifestFile)
+			if err != nil {
+				return errors.Wrap(err, "Unable to write to local manifest file")
+			}
+			writtenToManifest = true
 		}
 	}
 
-	result := &unstructured.Unstructured{}
-	if !instanceFound {
-		// Create Deployment
-		log.Infof("Creating %s...", gvk.Kind)
-		result, err = myclient.Resource(gvr).Namespace(namespace).Create(deploymentManifest, metav1.CreateOptions{})
-		//	result, err := a.Client.DynamicClient.Resource(gvr).Namespace(namespace).Create(deploymentManifest, metav1.CreateOptions{})
-	} else {
-		// Update Deployment
-		log.Infof("Updating %s...", gvk.Kind)
-		result, err = myclient.Resource(gvr).Namespace(namespace).Update(deploymentManifest, metav1.UpdateOptions{})
-	}
+	_ = manifestFile.Close()
```

Comment:
we would want to return this error. I would rather write a defer where we create the files as such:
```[golang]
defer func() {
      merr := manifestFile.Close()
      if err == nil {
            err = merr
      }
}()

Reply:
Agree